pymongo/collection.py

```diff
@@ -116,7 +117,7 @@ def __init__(
         database: "Database[_DocumentType]",
         name: str,
         create: Optional[bool] = False,
-        codec_options: Optional[CodecOptions] = None,
+        codec_options: Optional["CodecOptions[_DocumentTypeArg]"] = None,
```

Comment:
How come this one can be "CodecOptions" not "bson.CodecOptions"?

Reply:
I can't make heads or tails of it, other than the fact that this a constructor.  I tried removing the docstring from `with_options`.